ISPN-11479 Implement <indexed-entities> for protobuf payloads in indexed caches (#8271)
infinispan/infinispan: documentation/src/main/asciidoc/topics/upgrading.adoc

* The `auto-config` attribute is deprecated and will be removed in a future version.

> * Starting with version 11 it is mandatory to declare all indexed types in the indexing configuration or else warnings will be logged every time an undeclared type is indexed or queried.

Reviewer: ```suggestion
* Starting with version 11 it is mandatory to declare all indexed types in the indexing configuration or else warnings will be logged when the undeclared type is firstly used.
```

Author: Done!

> This requirement exists solely for indexed caches and applies to both Java classes and protobuf types.

Reviewer: this will flood the logs for sure :smile:
I wonder if it's possible to print only at cache creation

Author: I'll try to move that to ```once```

Reviewer: ```@Once``` will be once for all caches I'm afraid. Ideally it should per once per cache

Author: I added an internal flag in ProtobufValueWrapperFieldBridge to log only once per cache
